crates/data_components/src/dynamodb/mod.rs

```diff
@@ -67,3 +69,23 @@ pub enum Error {
     #[snafu(display("Table has no partition key"))]
     MissingPartitionKey,
 }
+
+#[derive(Debug, Snafu)]
+pub enum StreamError {
+    #[snafu(display("Failed to receive DynamoDB Stream record: {source}"))]
+    FailedToReceiveMessage {
+        source: dynamo_subscriber::error::Error,
+    },
+
+    #[snafu(display("Unable to downcast ArrayBuilder"))]
```

Comment:
nit: Not very public user friendly (i.e. people who don't know what a Rust downcast is).